client/dashboard/app/notifications/index.tsx

```diff
@@ -13,22 +13,52 @@ const AsyncNotificationApp = lazy( () => import( '@automattic/notifications/src/
 export default function Notifications( { className }: { className: string } ) {
 	const navigate = useNavigate();
 	const { user } = useAuth();
+	const [ isOpen, setIsOpen ] = useState( false );
 	const [ hasUnseenNotifications, setHasUnseenNotifications ] = useState( user.has_unseen_notes );
 
-	const actionHandlers = ( onClosePanel: () => void ) => ( {
+	const handleToggle = ( willOpen: boolean ) => {
+		setIsOpen( willOpen );
+	};
+
+	const handleClose = () => {
+		handleToggle( false );
+	};
+
+	const actionHandlers = {
 		APP_RENDER_NOTES: [
 			( store: any, { newNoteCount }: { newNoteCount: number } ) => {
 				setHasUnseenNotifications( newNoteCount > 0 );
 			},
 		],
 		VIEW_SETTINGS: [
 			() => {
-				onClosePanel();
+				handleClose();
 				navigate( { to: '/me/notifications' } );
 			},
 		],
-		CLOSE_PANEL: [ onClosePanel ],
-	} );
+		CLOSE_PANEL: [ handleClose ],
+	};
+
+	useEffect( () => {
+		const handleKeyDown = ( event: KeyboardEvent ) => {
+			if ( event.target !== document.body && ( event.target as HTMLElement )?.tagName !== 'A' ) {
```

Comment:
Why do we need to check the `a` element 🤔

Reply:
Hmm, you're right, I was just following existing 

https://github.com/Automattic/wp-calypso/blob/6804a71e4e88736113ca48bf61773039d95e2745/client/notifications/index.jsx#L237

But it seems to work without it, so OK I'm removing it!